internal/provider/provider_helpers.go

```diff
@@ -0,0 +1,186 @@
+package provider
+
+import (
+	"crypto/rsa"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mitchellh/go-homedir"
+	"github.com/snowflakedb/gosnowflake"
+	"github.com/youmark/pkcs8"
+	"golang.org/x/crypto/ssh"
+)
+
+func mergeSchemas(schemaCollections ...map[string]*schema.Resource) map[string]*schema.Resource {
```

Comment:
with new provider I would like to have unit tests for all such helper methods. Let's just keep it in mind, this should not be a part of this PR but let's not forget about this.

Reply:
agreed